staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -228,11 +228,13 @@ func PatchResource(r rest.Patcher, scope *RequestScope, admit admission.Interfac
 		}
 		trace.Step("Object stored in database")
 
-		if err := setObjectSelfLink(ctx, result, req, scope.Namer); err != nil {
-			scope.err(err, w, req)
-			return
+		if !utilfeature.DefaultFeatureGate.Enabled(features.RemoveSelfLink) {
```

Comment:
I will take a deeper look on Monday (I'm mostly OOO today).

That said - I'm wondering if we don't want to just wait for:
https://github.com/kubernetes/enhancements/issues/1164

[which we wanted to do in 1.24 - we probably should check with Clayton and others on that again - the initial plan was to GA in 1.21, but Clayton suggested to wait until 1.24 because it hit us in some places.]

And with those changes the changes in this PR would be much simpler and more obvious...

Reply:
As commented on slack - I suggest splitting this PR into the part that is universally useful and a second PR that will actually do the "deep copy removal".

Didn't yet have time to look deeper - will try to get to it soon (tomorrow?)